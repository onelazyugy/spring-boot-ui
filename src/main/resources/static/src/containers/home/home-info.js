import React, { Component } from 'react';

export default class HomeInfo extends Component {

    constructor(props, context) {
        super(props, context);
    }

    render() {
        return (
            <div>
                <div className="home-message flex-column">
                    <label className="home-title">React + Redux Framework</label>
                    <span>This framework contains the following features to help kickstart your product <i>(*=still in development)</i></span>
                </div>
                <div className="home-links flex-row">
                    <a className="button" href="http://ux.homedepot.com/style-guide/" target="_blank">View Style Guide</a>
                    <a className="button" href="https://github.homedepot.com/ux/react-framework" target="_blank">View GitHub Page</a>
                </div>
                <div className="home-features">
                    <div className="flex-column">
                        <img src="assets/react-redux.png" />
                        <label className="feature-header">React + Redux</label>
                        <label>React-Redux Router</label>
                        <label>Redux Thunk</label>
                        <label>Webpack</label>
                    </div>
                    <div className="flex-column">
                        <i className="icon_eux" />
                        <label className="feature-header">UX STANDARDS</label>
                        <label>Style Guide</label>
                        <label>Pattern Library</label>
                        <label>Analytics*</label>
                    </div>
                    <div className="flex-column">
                        <i className="icon_devices" />
                        <label className="feature-header">DESKTOP & MOBILE</label>
                        <label>Mobile First*</label>
                        <label>Repsonsive Design*</label>
                        <label>Cross Platform</label>
                    </div>
                    <div className="flex-column">
                        <img src="assets/pcf.png" />
                        <label className="feature-header">PIVOTAL CLOUD FOUNDRY</label>
                        <label>Node.js & Express.js Server</label>
                        <label>Generates Manifest Files</label>
                        <label>Multiple Environment Configs</label>
                    </div>
                    <div className="flex-column">
                        <i className="icon_bug" />
                        <label className="feature-header">TDD</label>
                        <label>Jest & Enzyme</label>
                        <label>Snapshot Testing</label>
                        <label>Test Coverage Reports</label>
                    </div>
                    <div className="flex-column">
                        <i className="icon_homedepot" />
                        <label className="feature-header">THD INTEGRATIONS</label>
                        <label>Single Sign On</label>
                        <label>Security Static Analysis*</label>
                        <label>CI/CD Workflow*</label>
                    </div>
                </div>
            </div>
        );
    }
}
