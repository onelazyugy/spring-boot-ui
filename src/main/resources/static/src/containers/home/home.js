import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Route } from 'react-router';
import Header from 'components/header/header';
import Menu from 'components/menu/menu';
import HomeInfo from './home-info';
import ComponentShowcase from 'containers/component-showcase/component-showcase';
import * as LoginActions from 'actions/loginActions';

export class Home extends Component {

    constructor(props, context) {
        super(props, context);

        this.store = context.store;

        let loginState = this.store.getState().login;
        if (loginState.userProfile) {
            this.userProfile = loginState.userProfile;
        } else if (sessionStorage.getItem('userProfile')) {
            this.userProfile = JSON.parse(atob(sessionStorage.getItem('userProfile')));
        } else {
            this.userProfile = null;
        }
    }

    componentDidMount() {
        let useLogin = this.store.getState().login.useLogin;
        if (useLogin === true) {
            let unsubscribe = this.store.subscribe(() => {
                let loginState = this.store.getState().login;
                let sessionProfile = sessionStorage.getItem('userProfile');
                if (loginState.userProfile === null && sessionProfile === null) {
                    unsubscribe();
                    this.props.history.push('/login');
                }
            });
        }
    }

    render() {
        return (
            <div className="home">
                <Header empty={this.userProfile === null} userProfile={this.userProfile} logout={this.props.actions.logoutUser} />
                <div className="home-layout navigation-wrapper">
                    <Menu />
                    <div className="home-content">
                        <Route exact path="/home" component={HomeInfo} />
                        <Route path="/home/components" component={ComponentShowcase} />
                    </div>
                </div>
            </div>
        );
    }
}

Home.contextTypes = {
    store: PropTypes.object
};

Home.propTypes = {
    actions: PropTypes.object.isRequired,
    history: PropTypes.object
};

const mapDispatchToProps = dispatch => ({
    actions: bindActionCreators(LoginActions, dispatch)
});

export default connect(null, mapDispatchToProps)(Home);
