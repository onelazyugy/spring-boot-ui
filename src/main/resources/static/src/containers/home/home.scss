.home {
    @extend .flex-column;
    background-color: $black12;
    height: 100%;
    color: $blackA11;
    overflow: hidden;

    .home-layout {
        height: 100%;

        .home-content {
            background-color: $black12;
            width: 100%;
            padding: 10px;
            overflow: auto;
            box-sizing: border-box;
            margin-top: 90px;

            .home-message {
                min-height: 70px;
                margin-top: 20px;
                align-items: center;
                justify-content: center;
                text-align: center;

                .home-title {
                    color: $orange;
                    font-weight: 700;
                    font-size: 30px;
                    line-height: 30px;
                    text-align: center;
                }

                @media screen and (max-width: 435px) {
                    min-height: 170px;
                }
            }

            .home-links {
                min-height: 80px;
                align-items: center;
                justify-content: center;
                flex-wrap: wrap;

                @media screen and (max-width: 435px) {
                    min-height: 200px;

                    .button {
                        margin-left: 0;
                        margin-right: 0;
                    }
                }
            }

            .home-features {
                width: 100%;
                min-height: 450px;
                display: grid;
                grid-template-columns: repeat(3, 1fr);
                grid-column-gap: 10px;
                grid-row-gap: 10px;

                > .flex-column {
                    justify-content: center;
                    align-items: center;
                }

                .feature-header {
                    color: $blue;
                    font-weight: 600;
                    font-size: 20px;
                    margin-top: 5px;
                }

                label {
                    text-align: center;
                }

                i {
                    font-size: 65px;
                    color: $black17;

                    &.icon_homedepot {
                        color: $orange;
                        background-color: $white;
                    }
                }

                img {
                    height: 65px;
                }

                @media screen and (max-width: 435px) {
                    min-height: 1400px;

                    .feature-column {
                        div {
                            margin-bottom: 60px;
                        }
                    }
                }
            }
        }
    }
}
