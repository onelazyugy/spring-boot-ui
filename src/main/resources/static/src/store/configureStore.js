import { createStore, compose, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import rootReducer from '../reducers';

const history = createBrowserHistory();
const routing = routerMiddleware(history);

function configureStoreProd(initialState) {
    const middlewares = [
        routing,
        thunk
    ];

    return createStore(
        connectRouter(history)(rootReducer),
        initialState,
        compose(
            applyMiddleware(...middlewares)
        )
    );
}

function configureStoreDev(initialState) {
    const middlewares = [
        reduxImmutableStateInvariant(),
        routing,
        thunk
    ];

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools
    const store = createStore(
        connectRouter(history)(rootReducer),
        initialState,
        composeEnhancers(
            applyMiddleware(...middlewares)
        )
    );

    if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextReducer = require('../reducers').default; // eslint-disable-line global-require
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}

const configureStore = process.env.NODE_ENV === 'production' ? configureStoreProd : configureStoreDev;

export default configureStore;
