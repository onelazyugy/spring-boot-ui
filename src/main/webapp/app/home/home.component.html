<div>
    <h1>Home Component</h1>
    <logo></logo>
    <search></search>
</div>