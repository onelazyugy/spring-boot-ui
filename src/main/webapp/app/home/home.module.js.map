{"version":3,"file":"home.module.js","sourceRoot":"","sources":["home.module.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,sCAAyC;AACzC,0CAA+C;AAI/C,yDAAuD;AACvD,+DAA6D;AAC7D,mDAAiD;AAqBjD,IAAa,UAAU;IAAvB;IAAwB,CAAC;IAAD,iBAAC;AAAD,CAAC,AAAzB,IAAyB;AAAZ,UAAU;IAnBtB,eAAQ,CAAC;QACN,YAAY,EAAE;YACV,eAAe;YACf,8BAAa;YACb,kCAAe;SAClB;QACD,OAAO,EAAC;YACJ,aAAa;YACb,eAAe;YACf,qBAAY,CAAC,QAAQ,CAAC;gBAClB,EAAC,IAAI,EAAE,MAAM,EAAE,SAAS,EAAE,8BAAa,EAAC;aAC3C,CAAC;SACL;QACD,OAAO,EAAE;YACL,8BAAa;YACb,kCAAe;SAClB;KACJ,CAAC;;GAEW,UAAU,CAAE;AAAZ,gCAAU"}