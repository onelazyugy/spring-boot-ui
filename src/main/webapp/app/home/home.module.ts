import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { LogoModule } from '../logo/logo.module';
import { SearchModule } from '../search/search.module';
import { LogoComponent } from '../logo/logo.component';
import { SearchComponent } from '../search/search.component';
import { HomeComponent } from './home.component';

@NgModule({
    declarations: [
        LogoComponent,
        SearchComponent
    ],
    imports:[
        RouterModule.forChild([
            {path: 'home', component: HomeComponent}
        ])
    ],
    exports: [
        LogoComponent,
        SearchComponent
    ]
})

export class HomeModule{}