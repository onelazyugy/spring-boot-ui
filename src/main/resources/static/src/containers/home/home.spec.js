import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'store/configureStore';
import { ConnectedRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { Home } from './home';
import Header from 'components/header/header';

describe('<Home />', () => {
    let actions, history, mountedWrapper, store, wrapper;

    beforeAll(() => {
        actions = { logoutUser: () => { } };
        history = createBrowserHistory();
        store = configureStore({
            login: {
                userProfile: {}
            }
        });
        mountedWrapper = mount(<ConnectedRouter history={history}><Home actions={actions} /></ConnectedRouter>, { context: { store } });
        wrapper = shallow(<Home actions={actions} />, { context: { store: store } });
    });

    it('should match its empty snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render <Header/> component', () => {
        expect(wrapper.find(Header)).toHaveLength(1);
    });

    it('should pass store.userProfile to Header userProfile prop', () => {
        let userProfileProp = mountedWrapper.find(Header).prop('userProfile');
        expect(userProfileProp).toEqual({});
    });

    it('should pass logoutUser action to Header logout prop', () => {
        let logoutProp = mountedWrapper.find(Header).prop('logout');
        expect(logoutProp).toEqual(actions.logoutUser);
    });

});
