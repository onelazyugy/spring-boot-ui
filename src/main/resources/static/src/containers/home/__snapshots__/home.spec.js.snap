// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match its empty snapshot 1`] = `
<div
  className="home"
>
  <Header
    empty={false}
    logout={[Function]}
    userProfile={Object {}}
  />
  <div
    className="home-layout navigation-wrapper"
  >
    <withRouter(Menu) />
    <div
      className="home-content"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/home"
      />
      <Route
        component={[Function]}
        path="/home/components"
      />
    </div>
  </div>
</div>
`;
